From 432a99a5eeaa7934a7c4c263053995cc87614984 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 23 May 2021 11:41:00 +0300
Subject: [PATCH] [SW-40142] fw_upd: disable according to flag

Until now the fit closure command was used just to set
pcie instance state disable flag. This patch is doing
the actual disabling of interface by checking this flag
in module_fwupd_check routine and doing nothing if
this flag is set. This mechanism is a security
requirement.

Change-Id: I91e2611f2c4c34f295a276218b61328663fea806
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 794a72f..dfdd53c 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -287,6 +287,9 @@
 {
 	enum cmd_check_res res = NOT_FOUND;
 
+	if (state->is_disabled)
+		return res;
+
 	/*
 	 * the error status response, involved
 	 * the creation of a FIT file.
